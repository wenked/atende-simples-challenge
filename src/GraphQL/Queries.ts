import { gql } from '@apollo/client';

export const VEHICLESNAMES_QUERY = gql`
	query {
		allVehicles {
			vehicles {
				name
				id
			}
		}
	}
`;

export const FILMSTITLES_QUERY = gql`
	query {
		allFilms {
			films {
				title
				id
			}
		}
	}
`;

export const PEOPLENAMES_QUERY = gql`
	query {
		allPeople {
			people {
				id
				name
			}
		}
	}
`;

export const ALLDATA_QUERY = gql`
	query {
		allFilms {
			films {
				episodeID
				title
				openingCrawl
				director
				producers
				releaseDate
				id
			}
		}
		allPeople {
			people {
				id
				name
				birthYear
				eyeColor
				gender
				hairColor
				height
				mass
				skinColor
				homeworld {
					name
				}
				species {
					name
				}
				created
			}
		}
		allVehicles {
			vehicles {
				id
				name
				model
				vehicleClass
				manufacturers
				costInCredits
				length
				crew
				passengers
				maxAtmospheringSpeed
				cargoCapacity
				consumables
			}
		}
	}
`;

export const FILMS_QUERY = gql`
	query {
		allFilms {
			films {
				episodeID
				title
				openingCrawl
				director
				producers
				releaseDate
				created
				edited
				id
			}
		}
	}
`;

export const PEOPLE_QUERY = gql`
	query {
		allPeople {
			people {
				id
				name
				birthYear
				eyeColor
				gender
				hairColor
				height
				mass
				skinColor
				homeworld {
					name
				}
				species {
					name
				}
				created
			}
		}
	}
`;

export const VEHICLES_QUERY = gql`
	query {
		allVehicles {
			vehicles {
				id
				name
				model
				vehicleClass
				manufacturers
				costInCredits
				length
				crew
				passengers
				maxAtmospheringSpeed
				cargoCapacity
				consumables
			}
		}
	}
`;
