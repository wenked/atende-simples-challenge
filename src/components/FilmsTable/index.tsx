import React from 'react';
import { filmInterface } from 'src/types';
import { Cell, Container } from './styles';

interface FilmsTableProps {
	filmsData: filmInterface[];
}

const FilmsTable: React.FC<FilmsTableProps> = ({ filmsData }) => {
	return (
		<Container>
			<Cell id='row-header'>Episode Id</Cell>
			<Cell id='row-header'>ID</Cell>
			<Cell id='row-header'>Title</Cell>
			<Cell id='row-header'>Director</Cell>
			<Cell id='row-header'>Opening Crawl</Cell>
			<Cell id='row-header'>Producers</Cell>
			<Cell id='row-header'>Release Date</Cell>
			{filmsData.map((film, i) => {
				return (
					<React.Fragment key={i}>
						<Cell data-title='Episode ID'>{film.episodeID}</Cell>
						<Cell data-title='ID'>{film.id}</Cell>
						<Cell data-title='Title'>{film.title}</Cell>
						<Cell data-title='Director'>{film.director}</Cell>
						<Cell data-title='Opening Crawl'>{film.openingCrawl}</Cell>
						<Cell data-title='Producers'>
							{film.producers.map((producer, i) => {
								return (
									<React.Fragment key={i}>
										{producer}
										<br />
									</React.Fragment>
								);
							})}
						</Cell>
						<Cell data-title='Release Date'>{film.releaseDate}</Cell>
					</React.Fragment>
				);
			})}
		</Container>
	);
};

export default FilmsTable;
