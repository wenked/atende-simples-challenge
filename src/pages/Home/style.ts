import styled from 'styled-components';

export const Container = styled.div`
	margin: 10px;
	padding: 20px;
`;

export const LoadingContainer = styled.div`
	display: flex;
	justify-content: center;
	align-items: center;
	text-align: center;
`;
