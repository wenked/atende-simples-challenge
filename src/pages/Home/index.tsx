import { CustomButton } from '@components/Layout/CustomButton/styles';

import { ListContainer } from '@components/Layout/ListContainer/styles';
import React, { useState } from 'react';

import { useQuery } from '@apollo/client';
import { ALLDATA_QUERY } from '@graphql/Queries';

import { FilmsTable } from '@components/FilmsTable';
import VehiclesTable from '@components/VehiclesTable';
import { PeopleTable } from '@components/PeopleTable';
import { Container } from './style';
import { LoadingContainer } from '@components/Layout/LoadingContainer/styles';

const Home: React.FC = () => {
	const { loading, error, data } = useQuery(ALLDATA_QUERY);
	const [showFilmData, setShowFilmData] = useState(false);
	const [showVehicleData, setShowVehicleData] = useState(false);
	const [showPeopleData, setShowPeopleData] = useState(false);

	return error ? (
		<h2>Failed to fetch data</h2>
	) : loading ? (
		<LoadingContainer>
			<h2>Loading...</h2>
		</LoadingContainer>
	) : (
		<ListContainer>
			<Container>
				<CustomButton
					onClick={() => {
						setShowFilmData(!showFilmData);
					}}>
					Films
				</CustomButton>
				{showFilmData && <FilmsTable filmsData={data.allFilms.films} />}
			</Container>

			<Container>
				<CustomButton
					onClick={() => {
						setShowPeopleData(!showPeopleData);
					}}>
					People
				</CustomButton>
				{showPeopleData && <PeopleTable peopleData={data.allPeople.people} />}
			</Container>
			<Container>
				<CustomButton
					onClick={() => {
						setShowVehicleData(!showVehicleData);
					}}>
					Vehicles
				</CustomButton>
				{showVehicleData && (
					<VehiclesTable vehiclesData={data.allVehicles.vehicles} />
				)}
			</Container>
		</ListContainer>
	);
};

export default Home;
