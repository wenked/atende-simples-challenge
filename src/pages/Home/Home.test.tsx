import React from 'react';
import { mount } from 'enzyme';
import Home from './index';
import { FilmsTable } from '@components/FilmsTable';
import { MockedProvider } from '@apollo/client/testing';
import { ALLDATA_QUERY } from '@graphql/Queries';
import { act } from 'react-dom/test-utils';
import {
	fireEvent,
	getByAltText,
	getByTestId,
	render,
} from '@testing-library/react';

const filmsDataa = [
	{
		episodeID: 1,
		title: 'sdsds',
		openingCrawl: 'adadasd',
		director: 'asdasdasdasd',
		producers: ['saddasd'],
		releaseDate: 'string',
		id: 'assdasdfasd',
	},
];

const mocks = {
	request: {
		query: ALLDATA_QUERY,
	},
	result: {
		data: {
			allFilms: {
				films: [
					{
						episodeID: 4,
						title: 'A New Hope',
						openingCrawl:
							"It is a period of civil war.\r\nRebel spaceships, striking\r\nfrom a hidden base, have won\r\ntheir first victory against\r\nthe evil Galactic Empire.\r\n\r\nDuring the battle, Rebel\r\nspies managed to steal secret\r\nplans to the Empire's\r\nultimate weapon, the DEATH\r\nSTAR, an armored space\r\nstation with enough power\r\nto destroy an entire planet.\r\n\r\nPursued by the Empire's\r\nsinister agents, Princess\r\nLeia races home aboard her\r\nstarship, custodian of the\r\nstolen plans that can save her\r\npeople and restore\r\nfreedom to the galaxy....",
						director: 'George Lucas',
						producers: ['Gary Kurtz', 'Rick McCallum'],
						releaseDate: '1977-05-25',
						id: 'ZmlsbXM6MQ==',
					},
				],
			},
			allPeople: {
				people: [
					{
						id: 1,
						name: 'Luke Skywalker',
						birthYear: '19BBY',
						eyeColor: 'blue',
						gender: 'male',
						hairColor: 'blond',
						height: 172,
						mass: 77,
						skinColor: 'fair',
						homeworld: {
							name: 'Tatooine',
						},
						species: null,
						created: '2014-12-09T13:50:51.644000Z',
					},
				],
			},
			allVehicles: {
				vehicles: [
					{
						id: 'dmVoaWNsZXM6NA==',
						name: 'Sand Crawler',
						model: 'Digger Crawler',
						vehicleClass: 'wheeled',
						manufacturers: ['Corellia Mining Corporation'],
						costInCredits: 150000,
						length: 36.8,
						crew: '46',
						passengers: '30',
						maxAtmospheringSpeed: 30,
						cargoCapacity: 50000,
						consumables: '2 months',
					},
				],
			},
		},
	},
};

describe('rendering component', () => {
	it('rendering  without crashing, testing buttons,rendering table', async () => {
		let wrapper = mount(
			<MockedProvider
				mocks={[mocks]}
				addTypename={false}
				defaultOptions={{ watchQuery: { fetchPolicy: 'no-cache' } }}>
				<Home />
			</MockedProvider>
		);

		await act(async () => {
			await new Promise((resolve) => setTimeout(resolve, 1000));
		});

		wrapper.update();

		console.log(wrapper.debug());
		wrapper.find('CustomButton').first().simulate('click');
		console.log(wrapper.debug());

		expect(wrapper.find('FilmsCell').first().text()).toBe('Episode Id');

		wrapper.find('CustomButton').at(1).simulate('click');
		expect(wrapper.find('PeopleCell').first().text()).toBe('ID');
		console.log(wrapper.debug());

		wrapper.find('CustomButton').at(2).simulate('click');
		console.log(wrapper.debug());
		expect(wrapper.find('VehicleCell').at(1).text()).toBe('Name');
	});
});

test('snapshot', async () => {
	const { container } = render(
		<MockedProvider
			mocks={[mocks]}
			addTypename={false}
			defaultOptions={{ watchQuery: { fetchPolicy: 'no-cache' } }}>
			<Home />
		</MockedProvider>
	);

	await act(async () => {
		await new Promise((resolve) => setTimeout(resolve, 1000));
	});

	expect(container.firstChild).toMatchSnapshot();
});

describe('passing props ', () => {
	const FilmsWrapper = mount(<FilmsTable filmsData={filmsDataa} />);
	it('accepts filmsData props', () => {
		expect(FilmsWrapper.props().filmsData).toEqual(filmsDataa);
	});
});
