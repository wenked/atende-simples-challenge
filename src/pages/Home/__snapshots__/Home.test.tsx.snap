// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  class="sc-gsTCUz uRvXt"
>
  <div
    class="sc-fubCfw dnDShV"
    id="filmsbutton"
  >
    <button
      class="sc-bdfBwQ cbDZcM"
      id="normal-button"
    >
      Films
    </button>
  </div>
  <div
    class="sc-fubCfw dnDShV"
  >
    <button
      class="sc-bdfBwQ cbDZcM"
    >
      People
    </button>
  </div>
  <div
    class="sc-fubCfw dnDShV"
  >
    <button
      class="sc-bdfBwQ cbDZcM"
    >
      Vehicles
    </button>
  </div>
</div>
`;
